app/src/main/java/org/dhis2/data/server/ServerModule.kt

```diff
@@ -26,12 +27,17 @@ import org.dhis2.utils.analytics.AnalyticsInterceptor
 import org.hisp.dhis.android.core.D2
 import org.hisp.dhis.android.core.D2Configuration
 import org.hisp.dhis.android.core.D2Manager
+import org.hisp.dhis.android.core.D2Manager.blockingInstantiateD2
 
 @Module
 class ServerModule {
     @Provides
     @PerServer
-    fun sdk(): D2 {
+    fun sdk(context: Context): D2 {
+        if (!D2Manager.isD2Instantiated()) {
+            val d2Configuration = blockingInstantiateD2(getD2Configuration(context))
+            d2Configuration!!.userModule().accountManager().setMaxAccounts(Constants.MAX_ACCOUNTS)
```

Comment:
why not `d2Cofiguration?.userModule()....` to make it null safe? what would happend if d1Configuration is null?